### DISCUSSION TITLE
#3 📑 Weekly Report 3

### DISCUSSION THREAD

**Discussion by fehlend0** [SE-TINF22B6/Statify#34] [Blog]
Hello everyone👋

Since the third week is almost over, here are the updates on the project.

## Activity Diagrams:
We designed three activity diagrams based on the use cases and added them to our project documentation.

[The first one](https://github.com/SE-TINF22B6/Statify/blob/c651b746693d3fd84825ad87fb295a99eb0a92a6/Documentation/Login_Activity_Diagram.svg) explaines, how the log-in process is meant to be, [the second one](https://github.com/SE-TINF22B6/Statify/blob/c651b746693d3fd84825ad87fb295a99eb0a92a6/Documentation/Edit_Profile_Activity_Diagram.svg) explaines the edition of the user's profile, and [the third one](https://github.com/SE-TINF22B6/Statify/blob/c651b746693d3fd84825ad87fb295a99eb0a92a6/Documentation/Generate_Activity_Diagram.svg) is meant to depict the process of generation of statistics.

These diagrams provide a clear visual representation of the processes and interactions within our system, making it easier for the team to understand how everything fits together.

## Sequence Diagrams:
In addition to the activity diagrams, we created two sequence diagrams that help illustrate the order of interactions between various components of our project.

[Sequence diagram](https://github.com/SE-TINF22B6/Statify/blob/main/Documentation/diagrams/Sequence_diagram.pdf)

These diagrams are now part of our documentation, enhancing our project's clarity and accessibility.

## Task Completion:
Our team successfully tackled [the task](https://github.com/SE-TINF22B6/Statify/discussions/32#18) of using the Spotify Web API to obtain access to the user's Spotify-account . This achievement opens up various possibilities for accessing and generating statistics based on the user's listening behavior.

It's been a productive week, and we're looking forward to the next steps in our project. If you have any questions or need further information, feel free to reach out. Let's keep up the good work!

Your Statify Team

**Comment by AndresMasis** [comment]
I like the approach that you followed, of drawing different diagrams. I think in some causes it is better because it lets you express more specific situations instead of having a very big diagram that is difficult to understand by its great complexity. Congratulations also by using the Spotify API. That gets you a step closer to your final result, and I am glad for you. I hope that you can keep working at this pace.

Cheers, CookPit.

**Comment by TadaXD20** [comment]
I love your activity diagrams. They are well structured and summarize all the core features. You have not gone into much detail and yet it is understable which component is responsible for which function. I also wanted to take a look at your sequence diagram but I couldn't find it in your documentation. Did you choose another format or will it be added in the future?

Tanina from Team Plapy
